project-koku/koku
# Improve PostgreSQL and minio container builds

## Description

Make the UID and GID of the `postgres` user in the container match those of the current user. This makes the PostgreSQL data files easy to manage from the container host. Build a PostgreSQL image with the correct UID, GID, and our entry point script baked in. The fewer bind mounts the better.

Use environment variables to configure the UID and GID of the minio container. There is an entry point script that sets up the user account, so we don't need to maintain a separate container file.

Move container files into `dev/containers` . This keeps the repo root clean and makes builds easier.



## Testing

1. Checkout Branch
2. Delete containers and local data

```
make docker-down
make delete-db
make delete-trino
make delete-trino-data
rm -rf pg_data/*
rm -rf .trino/parquet_data/.minio.sys
```
3. Start it all back up: `make docker-up-min`
4. Check that things came up ok `make docker-logs`
5. Load some data `make load-test-customer-data`

## Notes

I would like to move all the dev container related files and data to `dev/containers`.
